app/screens/security-screen/two-factor-authentication-verification-screen.tsx

```diff
@@ -0,0 +1,221 @@
+import { Input } from "react-native-elements"
+import { gql, useLazyQuery, useMutation } from "@apollo/client"
+import { RouteProp } from "@react-navigation/native"
+import { StackNavigationProp } from "@react-navigation/stack"
+import EStyleSheet from "react-native-extended-stylesheet"
+
+import { Screen } from "../../components/screen"
+import { RootStackParamList } from "@app/navigation/stack-param-lists"
+import { translate } from "../../i18n/translate"
+import { color } from "../../theme/color"
+import { toastShow } from "../../utils/toast"
+import { TwoFAVerificationType } from "."
+import { GET_TWO_FA_ENABLED } from "../../graphql/query"
+
+const SAVE_2FA = gql`
+  mutation save2FA($secret: String!, $token: String!) {
+    save2fa(secret: $secret, token: $token)
+  }
+`
+
+const DELETE_2FA = gql`
+  mutation delete2FA($token: String!) {
+    delete2fa(token: $token)
+  }
+`
+
+// const DELETE_2FA = gql`
+//   mutation twoFADelete($input: TwoFADeleteInput!) {
+//     twoFADelete(input: $input) {
+//       errors {
+//         message
+//       }
+//       success
+//     }
+//   }
+// `
+
+type Props = {
+  navigation: StackNavigationProp<RootStackParamList, "twoFAVerification">
+  route: RouteProp<RootStackParamList, "twoFAVerification">
+}
+
+export const TwoFactorAuthenticationVerificationScreen = ({
+  route,
+  navigation,
+}: Props): JSX.Element => {
+  const { twoFASecret, verificationType } = route.params
+
+  const [code, setCode] = useState<string>("")
+
+  const [save2FA, { loading: loadingSave2FA }] = useMutation(SAVE_2FA, {
+    fetchPolicy: "no-cache",
+  })
+
+  const [delete2FA, { loading: loadingDelete2FA }] = useMutation(DELETE_2FA, {
+    fetchPolicy: "no-cache",
+  })
+
+  const [fetchTwoFAEnabled] = useLazyQuery(GET_TWO_FA_ENABLED, {
+    fetchPolicy: "network-only",
+  })
+
+  const send = async () => {
+    if (loadingSave2FA || loadingDelete2FA) {
+      return
+    }
+
+    if (code.length !== 6) {
+      toastShow(translate("TwoFAVerificationScreen.need6Digits"))
+      return
+    }
+
+    try {
+      if (verificationType === TwoFAVerificationType.Delete) {
+        const { data } = await delete2FA({
+          variables: { token: code },
+        })
+
+        const success = data?.delete2fa
+        if (success) {
+          fetchTwoFAEnabled()
+          navigation.navigate("security")
+        } else {
+          toastShow(translate("TwoFAVerificationScreen.errorVerifyingCode"))
+        }
+      } else {
+        const { data } = await save2FA({
+          variables: { secret: twoFASecret, token: code },
+        })
+
+        const success = data?.save2fa
+        if (success) {
+          fetchTwoFAEnabled()
+          navigation.navigate("security")
+        } else {
+          toastShow(translate("TwoFAVerificationScreen.errorVerifyingCode"))
+        }
+      }
+    } catch (err) {
+      console.warn({ err })
+      toastShow(`${err}`)
```

Comment:
Can we be sure that the error surfaced to the user here will be friendly?

Reply:
The problem is we might need the more descriptive error to debug issues for the app. I think for common errors we except we should have checks and translations and then this as a fallback. Perhaps rap it in a translation that says "There was an unknown error: ".